structure/tree/avl_test.go

```diff
@@ -1,125 +1,125 @@
-package avl
+package tree
```

Comment:
This test will completely break when you change this to `tree_test`. Try it!

Reply:
The reason I mention this is because you have the Tree but you can access any of its values. Its quite pointless if you can't use this outside the package right?

Reply:
Sorry, I don't get it. I think we can use `avl` outside the package. Every member of `avl` is exported.